src/Template/Sentences/adopt.ctp

```diff
@@ -37,4 +37,10 @@
  * @link     https://tatoeba.org
  */
 
-$this->Menu->adoptButton($sentenceId, $owner);
+if ($acceptsJson) {
+    $result = $this->Sentences->getSentenceData($sentence);
+    echo json_encode($result);
```

Comment:
If possible, I think it would be better to avoid passing `$acceptJson` to the view and using a helper to format json data here. It’s not really the helper’s responsibility to select which data to return in the response. You could use the `_serialize` key from the controller like I did in `ExportsController.php`.

Reply:
Thank you for the pointers. I refactored the code for adoption. There will be other parts to refactor but I will handle that when I get to remove obsolete code from the old sentence design.

There was some issue with loading the `RequestHandler` component in the `SentencesController::beforeFilter`.

```
1) App\Test\TestCase\Controller\SentencesControllerTest::testEditSentence_doesntWorkForUnknownSentence

Failed asserting that response has header 'Location'.
```

Was not sure how to solve it so I just loaded the component only in the renderAdopt() function.